src/mlpack/tests/main_tests/perceptron_test.cpp

```diff
@@ -232,4 +238,186 @@ BOOST_AUTO_TEST_CASE(PerceptronMaxItrTest)
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensuring that the response size is checked
+ **/
+BOOST_AUTO_TEST_CASE(PerceptronWrongResponseSizeTest)
+{
+  constexpr int D = 2;
+  constexpr int N = 10;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  arma::Row<size_t> trainY; // response vector with wrong size
+
+  // 8 responses
+  trainY << 0 << 0 << 1 << 0 << 1 << 1 << 1 << 0 << endr;
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+
+  // labels for training data have wrong size. Should give runtime error
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensuring that absence of responses is checked.
+ */
+BOOST_AUTO_TEST_CASE(PerceptronNoResponsesTest)
+{
+  constexpr int N = 10;
+  constexpr int D = 1;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  SetInputParam("training", std::move(trainX));
+
+  // No labels for training data. Should give runtime error
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensuring that absence of training data is checked.
+ */
+BOOST_AUTO_TEST_CASE(PerceptronNoTrainingDataTest)
+{
+  arma::Row<size_t> trainY;
+  trainY << 1 << 1 << 0 << 1 << 0 << 0 <<endr;
+
+  SetInputParam("labels", std::move(trainY));
+
+  // No training data. Should give runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+ * Ensuring that test data dimensionality is checked when model is loaded.
+ */
+BOOST_AUTO_TEST_CASE(PerceptronWrongDimOfTestData2)
+{
+  constexpr int N = 10;
+  constexpr int D = 3;
+  constexpr int M = 15;
+
+  arma::mat trainX = arma::randu<arma::mat>(D, N);
+  arma::Row<size_t> trainY;
+
+  // 10 responses
+  trainY << 0 << 1 << 0 << 1 << 1 << 1 << 0 << 1 << 0 << 0 << endr;
+
+  SetInputParam("training", std::move(trainX));
+  SetInputParam("labels", std::move(trainY));
+
+  // training the model
+  mlpackMain();
+
+  // get the output model obtained after the training
+  PerceptronModel* model =
+      CLI::GetParam<PerceptronModel*>("output_model");
+
+  // reset the data passed
+  CLI::GetSingleton().Parameters()["training"].wasPassed = false;
+  CLI::GetSingleton().Parameters()["labels"].wasPassed = false;
+
+  // test data with Wrong dimensionality.
+  arma::mat testX = arma::randu<arma::mat>(D - 1, M);
+  SetInputParam("input_model", model);
+  SetInputParam("test", std::move(testX));
+
+  // wrong dimensionality of test data. Should give runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
 BOOST_AUTO_TEST_SUITE_END();
+
+
```

Comment:
Can you remove the extra lines at the end of the file please?

Reply:
Done